### PR TITLE
feat(eoffice): implement integrations via SPLP via Internal Sidebar Gateway

### PR DESCRIPTION
**Pull request by azophy**
## Overview

- add integrations to [Sidebar's Internal Gateway](https://github.com/jabardigitalservice/sidebar-internal-gateway)
- add model & migration to store integration status for each kab/kota
- add basic controller to access those integration status
  > this new endpoints is intended to be used by Sidebar services to request other eoffice applications. This differs from the /ecosystem/* endpoints which is the reverse (to be consumed by other eoffice applications)

## Todo Next
- [x] test new receive naskah endpoint effect to sikd-website inbox menu
- [ ] update frontend registrasi drafting luar → pairing ke hikmal
- [ ] update simpan registrasi drafting luar terkait metadata eoffice
- [ ] update trigger utk kirim naskah ke integrasi eoffice

## Evidence
project: SIDEBAR
title: feat(eoffice): implement integrations via SPLP via Internal Sidebar Gateway
participants: @azophy @fajarhikmal214 @samudra-ajri @indraprasetya154

### CODE REVIEW

**Comment by azophy** [comment]
this new endpoints is intended to be used by Sidebar services to request other eoffice applications. This differs from the `/ecosystem/*` endpoints which is the reverse (to be consumed by other eoffice applications)

File: src/database/migrations/2023_02_22_101616_create_eoffice_integrations_table.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+use Illuminate\Support\Facades\DB;
+
+class CreateEofficeIntegrationsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
```
[the remark this answers is not in the document]
Comment (azophy) [reply src/database/migrations/2023_02_22_101616_create_eoffice_integrations_table.php:15]: the code for inserting initial data is 30 lines, there are no way to get around this limitation. ignoring...

**Review by samudra-ajri — APPROVED** [review]

**Review by fajarhikmal214 — APPROVED** [review]

**Review by indraprasetya154 — APPROVED** [review]

File: src/app/Models/EofficeIntegration.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        return self::query()
+            ->where(compact('regional_code'))
+            ->where('is_active', true)
+            ->first();
+    }
+
+    public function getSplpPath($type)
+    {
+        return "/splp{$this->splp_parent_path}/{$type}";
```
[the remark this answers is not in the document]
Comment (azophy) [reply src/app/Models/EofficeIntegration.php:33]: nice question. ini contoh integrasi yg udah ada di staging sekarang buat ke kab. bandung:

```
/splp/eokabbandung/1/ekosistem/sifat-naskah
```

untuk join2 slash ini emang lebih rapih kalau dibuat pakai kode sih daripada main2 string gini. apa mending pakai `implode()` aja yak? 🤔

[the remark this answers is not in the document]
Comment (azophy) [reply src/app/Models/EofficeIntegration.php]: aku coba rapihin di sini nih mal: https://github.com/jabardigitalservice/office-services/pull/641/commits/9374fe37be18dc76284940a169a12bf9c01da0dd

**Pull request merged**